### PR TITLE
media-gfx/gimp: version bump to 2.10.14

### PR DESCRIPTION
**Pull request by Barracuda72**
Bug: https://bugs.gentoo.org/699064

### CODE REVIEW

**Comment by Barracuda72** [comment]
Just the same ebuild as with 2.10.14. Works for me.

**Comment by band-a-prend** [comment]
@Barracuda72 please, recheck the build process. It seems new gimp 2.10.14 requires
`>=media-libs/babl-0.1.72` and `>=media-libs/gegl-0.4.18:0.4[cairo]`.

Also please notice that since 28 october 2019 this package and some of it's dependencies are maintainer needed.

**Comment by a17r** [comment]
Actually, you check Changelog and buildsystem before even starting to build. Github is not the place to simply add copy-paste bumps.

**Comment by Barracuda72** [comment]
> @Barracuda72 please, recheck the build process. It seems new gimp 2.10.14 requires
> `>=media-libs/babl-0.1.72` and `>=media-libs/gegl-0.4.18:0.4[cairo]`.

Thanks, nice catch! Sorry for missing that. My system already had latest versions, so the build went smoothly and I didn't notice.

> Also please notice that since 28 october 2019 this package and some of it's dependencies are maintainer needed.

That's very sad...

> Actually, you check Changelog and buildsystem before even starting to build. Github is not the place to simply add copy-paste bumps.

That's so nice of you! My bad, I won't make this mistake never ever again in my whole life.

**Comment by Barracuda72** [comment]
Because ~x86 is missing in gegl and libnsgif (#13541 ), I removed this keyword from GIMP also.

**Comment by band-a-prend** [comment]
Fix QA issue (`RESTRICT="!test? ( test )"`) and patch to fix [bug 685210](https://bugs.gentoo.org/685210) (with duplicate [bug 691070](https://bugs.gentoo.org/691070)) => [gimp.zip](https://github.com/gentoo/gentoo/files/3816836/gimp.zip).
Build log with tests result => [build.log](https://github.com/gentoo/gentoo/files/3816837/build.log).

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 1]
```diff
@@ -0,0 +1,226 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+PYTHON_COMPAT=( python2_7 )
+GNOME2_EAUTORECONF=yes
+
+inherit eapi7-ver virtualx autotools gnome2 multilib python-single-r1
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: `multilib` is superfluous in an EAPI-6 ebuild.

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+	>=media-libs/babl-0.1.72
+	>=media-libs/gegl-0.4.18:0.4[cairo]
+	aalib? ( media-libs/aalib )
+	alsa? ( media-libs/alsa-lib )
+	aqua? ( x11-libs/gtk-mac-integration )
+	gnome? ( gnome-base/gvfs )
+	virtual/jpeg:0
+	jpeg2k? ( >=media-libs/openjpeg-2.1.0:2= )
+	>=media-libs/lcms-2.8:2
+	mng? ( media-libs/libmng )
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: please check if it links against a versioned libmng soname which means you would need to add a `:=` slot operator here.

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	>=media-libs/freetype-2.1.7
+	>=media-libs/harfbuzz-0.9.19
+	>=media-libs/gexiv2-0.10.6
+	>=media-libs/libmypaint-1.3.0:=
+	>=media-gfx/mypaint-brushes-1.3.0
+	>=media-libs/fontconfig-2.12.4
+	sys-libs/zlib
+	dev-libs/libxml2
+	dev-libs/libxslt
+	x11-themes/hicolor-icon-theme
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: does it really need that at build time as well? (`${RDEPEND}` is later inherited in `DEPEND`).

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+	>=app-text/poppler-data-0.4.7
+	>=media-libs/libpng-1.6.25:0=
+	python?	(
+		${PYTHON_DEPS}
+		>=dev-python/pygtk-2.10.4:2[${PYTHON_USEDEP}]
+		>=dev-python/pycairo-1.0.2[${PYTHON_USEDEP}]
+	)
+	>=media-libs/tiff-3.5.7:0
+	>=gnome-base/librsvg-2.40.6:2
+	webp? ( >=media-libs/libwebp-0.6.0 )
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: please check if it links against a versioned libwebp soname which means you would need to add a := slot operator here.

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+	wmf? ( >=media-libs/libwmf-0.2.8 )
+	net-libs/glib-networking[ssl]
+	x11-libs/libXcursor
+	sys-libs/zlib
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: this is a duplicate of line 35, one reason why alphanumerical sorting is a good idea in general

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+	sys-apps/findutils
+	virtual/pkgconfig
+	>=dev-util/intltool-0.40.1
+	>=sys-devel/gettext-0.19
+	doc? ( >=dev-util/gtk-doc-1 )
+	>=sys-devel/libtool-2.2
+	>=sys-devel/automake-1.11
+	dev-util/gtk-doc-am"
+
+DOCS="AUTHORS ChangeLog* HACKING NEWS README*"
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: `DOCS` should be an array, but that means you can't use `*` globbing anymore. In my opinion, `ChangeLog` is enough as none of the history versions exist in Gentoo repo for a long time. `README*` only gives `README.i18n` in addition.

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: please move it below `IUSE` for better readability

File: media-gfx/gimp/gimp-2.10.14.ebuild [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+
+src_configure() {
+	_adjust_sandbox
+
+	local myconf=(
+		GEGL="${EPREFIX}"/usr/bin/gegl-0.4
+		GDBUS_CODEGEN="${EPREFIX}"/bin/false
+
+		--enable-default-binary
+		--disable-silent-rules
```
Comment (a17r) [inline media-gfx/gimp/gimp-2.10.14.ebuild]: you can drop this line, already added automatically by `>=EAPI-5`.

**Comment by Barracuda72** [comment]
Wow, so much stuff! Thanks for detailed review!
Unfortunately I don't have much time; as I know, @band-a-prend wants to take maintainership over this package and in next few days plans to create new PR. I'll leave everything to him.

**Pull request closed without merging**